### PR TITLE
fixed: the assumption of toolchain = crosscompile cause issues

### PR DESCRIPTION
**Pull request by notspiff**
ref https://github.com/gentoo/gentoo/pull/3027

### CODE REVIEW

**Comment by hudokkow** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
we should replace all occurrences of CMAKE_CROSSCOMPILING with KODI_CROSSCOMPILING in this case

**Comment by akva2** [comment]
i don't think you want to do that, as cmake can be crosscompiling without depends ?

**Comment by wsnipex** [comment]
It can, but I'm 95% sure it would fail and I'm not prepared to support that at this stage of the cmake build sys.
The "issue" is that we assume kodi depends whenever we use CMAKE_CROSSCOMPILING

**Comment by akva2** [comment]
could be. at least this allows me to build with clang without hassle :)

File: tools/depends/target/Toolchain.cmake.in [hunk 1]
```diff
@@ -80,3 +80,5 @@ set(ENV{PKG_CONFIG_LIBDIR} @prefix@/@deps_dir@/lib/pkgconfig)
 if(NOT CORE_SYSTEM_NAME STREQUAL linux)
   set(ADDONS_PREFER_STATIC_LIBS ON)
 endif()
+
+set(KODI_CROSSCOMPILING 1)
```
Comment (Memphiz) [inline tools/depends/target/Toolchain.cmake.in]: Isn't that assumption wrong too? Jenkins builds linux64 incl. Depends and it is a native compile.

**Comment by akva2** [comment]
No. Option just has a bad name i guess, it means kodi jail provided toolchain file that may or may not be a cross compile depending on platform. I chose a shorter name ;)

File: tools/depends/target/Toolchain.cmake.in (hunk 1 above)
Comment (wsnipex) [inline tools/depends/target/Toolchain.cmake.in]: even though osx uses the native compiler it's still a cross compile in every other sense. You could simply switch the target arch to 32bit (as we had some time ago) and it'd build without any other change.

**Comment by fetzerch** [comment]
Had a quick look, should we call it KODI_DEPENDSCOMPILE or sth like that? KODI_CROSSCOMPILING is fine by me as well, but might be confusing as memphiz pointed out.

There are a few more places that need to use the new var as well:
- project/cmake/scripts/common/ArchSetup.cmake:100 should print a message for the new var additionally
- project/cmake/CMakeLists.txt:96 should probably use the new var
- All occurances in project/cmake/modules need to use the new var, as they use variables that are only defined in our toolchain file

**Comment by akva2** [comment]
renamed KODI_DEPENDSBUILD and (tried) to address everything requested.

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by stefansaraev** [comment]
this does not feel right.

```
[  3%] Generating ServiceDescription.h
/bin/sh: JsonSchemaBuilder::JsonSchemaBuilder: command not found
build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/build.make:65: recipe for target 'build/ServiceDescription.h' failed
make[2]: *** [build/ServiceDescription.h] Error 127
CMakeFiles/Makefile2:3414: recipe for target 'build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/all' failed
make[1]: *** [build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

cross-copiling for aarch64 on x86_64 host. not using depends, I do already have JsonSchemaBuilder (done as part of kodi native compile step) in toolchain.

reverting the change to FindJsonSchemaBuilder.cmake makes it work for me again. http://sprunge.us/TALY

**Comment by wsnipex** [comment]
right, cornercases :)
we should probably make it KODI_DEPENDSBUILD OR CMAKE_CROSSCOMPILING
mind PR-ing?

**Comment by stefansaraev** [comment]
done

**Comment by maxnet** [comment]
libdvd also still messed up.

With KODI_DEPENDSBUILD it will search for an existing library.
https://github.com/xbmc/xbmc/blob/master/project/cmake/modules/FindLibDvd.cmake#L2

Without it will build library, but does not set the correct C compiler unless KODI_DEPENDSBUILD is set...
https://github.com/xbmc/xbmc/blob/master/project/cmake/modules/FindLibDvd.cmake#L79

**Comment by cjmayo** [comment]
Can this be backported to Krypton? Working for me on kodi-17.0_b6.

**Pull request merged**